src/Mobs/PathFinder.cpp

```diff
@@ -0,0 +1,233 @@
+
+
+#include "Globals.h"
+#include "PathFinder.h"
+#include "../Chunk.h"
+
+cPathFinder::cPathFinder(double a_MobWidth, double a_MobHeight)
+	: m_Path(nullptr)
+	, m_GiveUpCounter(0)
+	, m_TicksSinceLastPathReset(1000)
+{
+	m_Width = a_MobWidth;
+	m_Height = a_MobHeight;
+
+}
+
+
+
+
+
+ePathFinderStatus cPathFinder::GetNextWayPoint(cChunk &a_Chunk, Vector3d a_Source, Vector3d a_Destination, Vector3d &a_OutputWaypoint)
+{
+	m_FinalDestination = a_Destination;
+	m_Source = a_Source;
+
+	if (m_TicksSinceLastPathReset < 1000)
+	{
+		// No need to count beyond 1000. 1000 is arbitary here.
+		++m_TicksSinceLastPathReset;
+	}
+
+	/* if (ReachedFinalDestination())
+	{
+		StopMovingToPosition();
+		return false;
+	} */
+
+	if ((m_FinalDestination - m_PathDestination).Length() > 0.25)  // if the distance between where we're going and where we should go is too big.
```

Comment:
Why not use SqrLength? `(m_FinalDestination - m_PathDestination).SqrLength() > 0.25 * 0.25` is faster and just as clear.